### PR TITLE
[8.0][ADD][operating unit]

### PR DESCRIPTION
**Pull request by JordiBForgeFlow**
This is the backport to 8.0 of module 'operating_unit' from 9.0. Note that the module was approved for 9.0 here: https://github.com/OCA/operating-unit/pull/1 and merged. No change has been done in the backport.

### CODE REVIEW

**Comment by JordiBForgeFlow** [comment]
Seems that runbot was killed for some reason unknown to me.

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
+language: python
```
Comment (elicoidal) [inline .travis.yml:1]: remove?

File: README.md [hunk 2]
```diff
@@ -19,7 +19,7 @@ This part will be replaced when running the oca-gen-addons-table script from OCA
```
Comment (elicoidal) [inline README.md:19]: This is made by automatic script

File: operating_unit/README.rst [hunk 3]
```diff
@@ -0,0 +1,81 @@
+.. image:: https://img.shields.io/badge/license-LGPLv3-blue.svg
+   :target: https://www.gnu.org/licenses/lgpl.html
+   :alt: License: LGPL-3
+
+==============
+Operating Unit
+==============
+
+An operating unit (OU) is an organizational entity part of a company, with
```
Comment (elicoidal) [inline operating_unit/README.rst:9]: Operating Unit

File: operating_unit/README.rst [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+separate management ownership. Management by OU is aimed to introduce the
+following features:
+
+* Data partitioning from other OU.
+* Own sequencing schemes.
+* User access administration for data processing and reporting.
+* Product and customer independence of OU.
```
Comment (elicoidal) [inline operating_unit/README.rst:16]: s/of/per

File: operating_unit/README.rst [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+
+* Eficent Business and IT Consulting Services S.L. <contact@eficent.com>
+* Serpent Consulting Services Pvt. Ltd. <support@serpentcs.com>
+
+Maintainer
+----------
+
+.. image:: https://odoo-community.org/logo.png
+   :alt: Odoo Community Association
+   :target: http://odoo-community.org
```
Comment (elicoidal) [inline operating_unit/README.rst]: https (x2)

File: operating_unit/__openerp__.py [hunk 6]
```diff
@@ -0,0 +1,30 @@
+# -*- coding: utf-8 -*-
+# © 2015 Eficent Business and IT Consulting Services S.L.
+# - Jordi Ballester Alomar
+# © 2015 Serpent Consulting Services Pvt. Ltd. - Sudhir Arya
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl.html).
+
+{
+    "name": "Operating Unit",
+    "summary": "An operating unit (OU) is an organizational entity part of a "
```
Comment (elicoidal) [inline operating_unit/__openerp__.py:9]: "summary": "Organize and segregate all data through Operating Units, internal organizational entity of a "
-               "company",
  +

File: operating_unit/__openerp__.py [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+               "company",
+    "version": "8.0.1.0.0",
+    "author": "Eficent Business and IT Consulting Services S.L., "
+              "Serpent Consulting Services Pvt. Ltd.,"
+              "Odoo Community Association (OCA)",
+    "website": "http://www.eficent.com",
+    "category": "Generic",
```
Comment (elicoidal) [inline operating_unit/__openerp__.py]: not sure about that ("too generic"?). Accounting and finance seems a better fit even if reductive

File: operating_unit/models/operating_unit.py [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+    code = fields.Char('Code', required=True)
+    active = fields.Boolean('Active', default=True)
+    company_id = fields.Many2one(
+        'res.company', 'Company', required=True, default=lambda self:
+        self.env['res.company']._company_default_get('account.account'))
+    partner_id = fields.Many2one('res.partner', 'Partner', required=True)
+
+    _sql_constraints = [
+        ('code_company_uniq', 'unique (code,company_id)',
+         'The code of the operating unit must '
```
Comment (elicoidal) [inline operating_unit/models/operating_unit.py:24]: capitals

File: operating_unit/security/operating_unit_security.xml [hunk 9]
```diff
@@ -0,0 +1,18 @@
...
+<openerp>
+<data noupdate="1">
+
+    <record id="group_multi_operating_unit" model="res.groups">
+        <field name="name">Multi Operating Unit</field>
+        <field name="category_id" ref="base.module_category_usability"/>
+    </record>
+
+    <record id="operating_unit_comp_rule" model="ir.rule">
+        <field name="name">Operating Unit Entry</field>
```
Comment (elicoidal) [inline operating_unit/security/operating_unit_security.xml:11]: is it "data entry"?

Comment (elicoidal) [inline operating_unit/security/operating_unit_security.xml:11]: I miss Operating Unit Manager (responsible for setup)

File: README.md [hunk 10]
```diff
@@ -19,7 +19,7 @@ This part will be replaced when running the oca-gen-addons-table script from OCA
 
 Translation Status
 ------------------
-[![Transifex Status](https://www.transifex.com/projects/p/OCA-operating-unit-8-0/chart/image_png)](https://www.transifex.com/projects/p/OCA-operating-unit-8-0)
+[![Transifex Status](https://www.transifex.com/projects/p/OCA-operating-unit-9-0/chart/image_png)](https://www.transifex.com/projects/p/OCA-operating-unit-9-0)
```
Comment (dreispt) [inline README.md]: This PR if for 8.0, so this is incorrect.

File: operating_unit/security/operating_unit_security.xml (hunk 9 above)
Comment (JordiBForgeFlow) [inline operating_unit/security/operating_unit_security.xml:11]: As it happens with Companies, only ERP Manager group is able to set up an OU. I think is good to keep the same approach as the company model

Comment (JordiBForgeFlow) [inline operating_unit/security/operating_unit_security.xml:11]: What do you mean by "is it data entry"? The ir.rule operating_unit_comp_rule is meant to restrict the access to the OU's only to users that have access to the associated company.

File: operating_unit/README.rst (hunk 4 above)
Comment (JordiBForgeFlow) [inline operating_unit/README.rst:16]: Hmm, not sure that your review is correct. http://forum.wordreference.com/threads/to-be-independent-of-from-with.1116616/

File: operating_unit/security/operating_unit_security.xml (hunk 9 above)
Comment (elicoidal) [inline operating_unit/security/operating_unit_security.xml:11]: I dont understand: "Operating Unit Entry"
Is it for data entry? or Accounting Entry?

File: operating_unit/README.rst (hunk 4 above)
Comment (elicoidal) [inline operating_unit/README.rst:16]: Looks good to me "independent of" but not sure "something independence of something else" (rather have from or per).
Just a detail ! and not being a native speaker, I let you make the decision :)

File: operating_unit/security/operating_unit_security.xml (hunk 9 above)
Comment (JordiBForgeFlow) [inline operating_unit/security/operating_unit_security.xml:11]: It is related to creating/editing an operating unit. Only a user that belongs to the company of the OU can access to it.

**Comment by JordiBForgeFlow** [comment]
Seems that Travis has crashed.

**Comment by dreispt** [comment]
I think TravisCi is not working because its config files are added by this PR.
:+1:

**Comment by JordiBForgeFlow** [comment]
@dreispt So perhaps I should create a PR with just the config files, right?

**Comment by dreispt** [comment]
@jbeficent That would be the best way, but IMO not worth the trouble now.

**Comment by JordiBForgeFlow** [comment]
@moylop260 Seems that there is an issue with Travis with this repository. There is an error "The command "travis_install_nightly" failed and exited with 1 during ." Can you help us?

**Comment by moylop260** [comment]
Hello @jbeficent 
Maybe is a copy&paste issue, check the first line of the following file: [sample_files/.travis.yml#L1](https://github.com/OCA/maintainer-quality-tools/blob/620b87395086ba546c50339601eb0f21236cc3a1/sample_files/.travis.yml#L1)

`language: python` is missing in your `.travis.yml` file

File: .travis.yml [hunk 11]
```diff
@@ -0,0 +1,37 @@
+sudo: false
```
Comment (moylop260) [inline .travis.yml:2]: Please, add `language: python` in the first line of this file

File: .travis.yml [hunk 12]
```diff
@@ -0,0 +1,37 @@
...
+cache: pip
```
Comment (moylop260) [inline .travis.yml:3]: Please, use the [cache with directories](https://github.com/OCA/maintainer-quality-tools/blob/620b87395086ba546c50339601eb0f21236cc3a1/sample_files/.travis.yml#L3-L6)

File: .travis.yml [hunk 13]
```diff
@@ -0,0 +1,37 @@
...
+
+addons:
+  apt:
+    packages:
+      - expect-dev  # provides unbuffer utility
+      - python-lxml  # because pip installation is slow
+
+env:
+  global:
+  - VERSION="8.0" LINT_CHECK="1"
```
Comment (moylop260) [inline .travis.yml]: In `global` section use just one line with:
`- VERSION="8.0" LINT_CHECK="0" TESTS="0" TRANSIFEX="0"`

Check the [global sample](https://github.com/OCA/maintainer-quality-tools/blob/620b87395086ba546c50339601eb0f21236cc3a1/sample_files/.travis.yml#L33-L64)

**Comment by AaronHForgeFlow** [comment]
Thank you @moylop260. All checks have passed but there's still an empty folder which I added by mistake. I tried to rebase but it did not work. How can I delete this folder?

**Comment by AaronHForgeFlow** [comment]
@dreispt @sbidoul I think this is ready to be merged. Note that this is the backport to 8.0 of module 'operating_unit' from 9.0 that was approved here: [https://github.com/OCA/operating-unit/pull/1](url)

File: operating_unit/models/operating_unit.py [hunk 14]
```diff
@@ -0,0 +1,42 @@
...
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+from openerp import api, fields, models
+
+
+class OperatingUnit(models.Model):
+
+    _name = 'operating.unit'
+    _description = 'Operating Unit'
+
+    name = fields.Char('Name', required=True)
```
Comment (moylop260) [inline operating_unit/models/operating_unit.py]: If you field name is equal to you string field you could remove the fiel string.
Odoo use the field name by default as string.

`s/'Name',//g`

**Comment by moylop260** [comment]
Please check the beta lint messages from first build
https://travis-ci.org/OCA/operating-unit/jobs/144127499#L420

File: operating_unit/models/operating_unit.py (hunk 14 above)
Comment (moylop260) [inline operating_unit/models/operating_unit.py]: Based on https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md#field

**Comment by moylop260** [comment]
:+1:

**Pull request merged**